pkg/karmadactl/top/top_pods.go

```diff
@@ -171,39 +176,71 @@ func (o *TopPodOptions) RunTopPod() error {
 		}
 	}
 
-	apiGroups, err := o.DiscoveryClient.ServerGroups()
+	o.Clusters, err = GenClusterList(o.karmadaClient, o.Clusters)
 	if err != nil {
 		return err
 	}
 
-	metricsAPIAvailable := SupportedMetricsAPIVersionAvailable(apiGroups)
+	wg.Add(len(o.Clusters))
+	for _, cluster := range o.Clusters {
+		go o.getPodMetrics(&wg, &mux, f, cluster, labelSelector, fieldSelector, &allErrs)
+	}
+	wg.Wait()
```

Comment:
How about following code:
```go
	wg.Add(len(o.Clusters))
	for _, cluster := range o.Clusters {
		go func() {
			defer wg.Done()
			o.getPodMetrics(&wg, &mux, f, cluster, labelSelector, fieldSelector, &allErrs)
		}
	}
	wg.Wait()
```

To provide clearer context oof wg.

Reply:
Good idea! Moving ```wg.done()``` out also makes the function's responsibilities more focused. I will modify pr #4224 simultaneously.